setup.py

```diff
@@ -875,27 +875,27 @@ def swig_sources(self, sources, ext=None):
 
 class my_install(install):
     def run(self):
+        """Custom script we run at the end of installing"""
         install.run(self)
-        # Custom script we run at the end of installing - this is the same script
-        # run by bdist_wininst
-        # This child process won't be able to install the system DLLs until our
-        # process has terminated (as distutils imports win32api!), so we must use
-        # some 'no wait' executor - spawn seems fine!  We pass the PID of this
-        # process so the child will wait for us.
-        # XXX - hmm - a closer look at distutils shows it only uses win32api
-        # if _winreg fails - and this never should.  Need to revisit this!
-        # If self.root has a value, it means we are being "installed" into
-        # some other directory than Python itself (eg, into a temp directory
-        # for bdist_wininst to use) - in which case we must *not* run our
-        # installer
-        if not self.dry_run and not self.root:
+        # If self.root has a value, it means we are being "installed" into some other
+        # directory than Python itself - in which case we must *not* run our installer.
+        # bdist_wininst used to trigger this by using a temp directory.
+        # Is this still a concern ?
+        if self.root:
+            print(
+                f"Not executing post install script when not installing in Python itself (self.root={self.root})"
+            )
+        if not self.dry_run:
             # We must run the script we just installed into Scripts, as it
             # may have had 2to3 run over it.
             filename = os.path.join(self.install_scripts, "pywin32_postinstall.py")
             if not os.path.isfile(filename):
                 raise RuntimeError(f"Can't find '{filename}'")
+            if not self.install_lib:
+                raise RuntimeError(f"Missing destination (install_lib is empty)")
             print("Executing post install script...")
-            # What executable to use?  This one I guess.
+            # As of setuptools>=74.0.0, we no longer need to
+            # be concerned about distutils calling win32api
```

Comment:
> As of setuptools>=74.0.0, we no longer need to be concerned about distutils calling win32api

See https://github.com/pypa/setuptools/pull/4606/files#diff-a40c9a474ee3b73d97a6934b5700e8d2931e4e2b233645448a3e5a145d88954c

>  We must run the script we just installed into Scripts, as it may have had 2to3 run over it.

That's no longer relevant, but I'll open a separate PR for the leftover lib2to3 bits (similar to #2311)

Given the postinstall script depends on pywin32 already being installed, it might be cleaner to keep it as a script/separate process call rather than importing it here. Will see.